community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -854,6 +854,16 @@ public PrimitiveLongIterator nodesGetFromIndexContainsScan( KernelStatement stat
         return filterIndexStateChangesForScanOrSeek( state, index, null, committed );
     }
 
+    @Override
+    public PrimitiveLongIterator nodesGetFromIndexEndsWithScan( KernelStatement state, IndexDescriptor index,
+            String suffix ) throws IndexNotFoundKernelException
+    {
+        StorageStatement storeStatement = state.getStoreStatement();
+        IndexReader reader = storeStatement.getIndexReader( index );
+        PrimitiveLongIterator committed = reader.endsWith( suffix );
+        return filterIndexStateChangesForScanOrSeek( state, index, null, committed );
```

Comment:
`filterIndexStateChangesForScanOrSeek` can be changed to more optimised version that will consider only properties that ends with specified suffix. As example `nodesGetFromIndexRangeSeekByPrefix` using specific `filterIndexStateChangesForRangeSeekByPrefix`